datalad/dataset/gitrepo.py

```diff
@@ -541,7 +541,7 @@ def init(self, sanity_checks=True, init_options=None):
         pathobj = self.pathobj
         path = str(pathobj)
 
-        if not lexists(path):
+        if not (pathobj.exists() or pathobj.is_symlink()):
```

Comment:
`lexists()` essentially is `os.path.stat(follow_symlinks=False)` and returning `True` unless an exception occurs. The single-operation `pathlib` equivalent would be `stat(follow_symlinks=False)`. The patch here, runs two operations against the file-system, instead of one -- which is likely going to be slower.

Reply:
> The single-operation pathlib equivalent would be stat(follow_symlinks=False)

Can't see that working. `Path.stat()` doesn't take any arguments.

Reply:
> Changed in version 3.10: The follow_symlinks parameter was added.

I guess your "Can't see that working" is referring to replacing your code with a `pathlib.stat()` call. My motivation for looking into this was

> Which seems wild, since the mkdir call apparently causing the FileExistsError is is done right after testing for lexists to be False.
> Resolved by using pathlib: 6f86a1b

I cannot see how the change you made could be causal re a fix. The primary difference is one vs two calls to `stat()`, which implies a timing difference which could be indicative of a race condition. Especially given the observed behavior of a random element in when exactly things fail.

FTR: `exists()` is `stat()` and `is_symlink()` is `S_ISLNK(self.lstat().st_mode)`

Reply:
1. Thx for `lstat` - I missed that.
2. Despite going for a simple `try: pathobj.lstat() except: FileNotFoundError` the issue seems gone.

I do agree that it looks awfully like a race condition, though - just no idea yet, why it would be observable only herein on that workflow. I'll try and restart a few times to see whether I can still trigger it.

I guess, `pathlib` makes still sense to use, but will move to a `pathobj_lexists` or similar.

Reply:
Currently back to using `os.path.lexists(pathlib.Path)`. Still not triggered again.
At this point I'm very confident it has nothing to do with this PR, but with some parallel usecases involving `ProducerConsumer` (something's ack of thread-safety?).
Moved to https://github.com/datalad/datalad/issues/6598